xchange-binance/src/main/java/org/knowm/xchange/binance/service/BinanceTradeService.java

```diff
@@ -367,9 +363,52 @@ public OpenOrdersParams createOpenOrdersParams() {
                             .getExchangeSpecificParametersItem("recvWindow"),
                     getTimestamp())));
       }
-      return orders;
+
+      // Get fees
+      return addFees(orders);
     } catch (BinanceException e) {
       throw BinanceErrorAdapter.adapt(e);
     }
   }
+
+  private Collection<Order> addFees(Collection<Order> orders) {
+    List<BinanceTrade> binanceTrades = new ArrayList<>();
+    orders.stream()
+        .map(Order::getCurrencyPair)
+        .distinct() //Optimization to reduce overhead
+        .forEach(
+            c -> {
+              try {
+                binanceTrades.addAll(
+                    super.myTrades(
+                        c,
+                        null,
+                        null,
+                        null,
+                        null,
+                        (Long)
+                            exchange
+                                .getExchangeSpecification()
+                                .getExchangeSpecificParametersItem("recvWindow"),
+                        getTimestamp()));
+              } catch (IOException e) {
```

Comment:
this is like the worst way you could handle an error
leaving the fields you think should always be filled null and a stack trace outside of the logging system

Reply:
I wrote this up fairly quickly to see if it would work. Please review the changes again as I've rectified the issues.